File: Low Level Design (LLD)/Problems/ATM/Java/Bank.java
import java.util.HashMap;
import java.util.Map;

public class Bank {
    private final Map<String, User> users = new HashMap<>();

    public void addUser(User user) {
        users.put(user.getCard().getCardNumber(), user);
    }

    public User authenticate(String cardNumber, String pin) {
        User user = users.get(cardNumber);
        if (user != null && user.getCard().validatePin(pin)) {
            return user;
        }
        return null;
    }
}

File: Low Level Design (LLD)/Problems/ATM/Cpp/src/Bank.cpp
#include "../include/Bank.h"

void Bank::addUser(User* user) {
    users[user->getCard()->getCardNumber()] = user;
}

User* Bank::authenticate(const std::string& cardNumber, const std::string& pin) {
    auto it = users.find(cardNumber);
    if (it != users.end() && it->second->getCard()->validatePin(pin)) {
        return it->second;
    }
    return nullptr;
}